### ISSUE TITLE
Testing PHP file after download results in error since there is no logfile created yet

### ISSUE THREAD

**Issue by hgonzago** [Esri/resource-proxy#11]
Downloaded the ZIP, copied PHP folder the Apache web server folder, made sure read/write permissions were set where appropriate. Following steps in PHP README.md file where it states to test the proxy was installed correctly by typing in

http://[yourmachine]/PHP/proxy.php?http://services.arcgisonline.com/ArcGIS/rest/services/?f=pjson

results in an error "{"error":{"code":412,"details":["Detected malformed 'logFile' in configuration.  Make sure this app has write permissions to specified log file in configuration.  The server does not meet one of the preconditions that the requester put on the request."],"message":"Proxy failed due to configuration error."}}"

The error goes away as soon as I manually create the proxy_log.log file.

**Comment by kodokpintar** [comment]
I got that exactly error when I did ping test. I am using IIS 7 and there is no issue on permission. Can anyone give advise?

Thanks.

**Comment by jgravois** [comment]
@kodokpintar did you manually create a proxy_log.log file?

**Comment by phpmaps** [comment]
@kodokpintar when you say "there is no issue on permission" would you take a another troubleshooting step and share the name of the folder that that proxy.php file lives in (so we have something concrete to talk about and then describe what type of permissions are on the folder?   The proxy should create the file for you if you have the correct permission... doing what @jgravois  says should work to and yield something like this 
 
```
01-28-15 15:33:33 | GET detected
01-28-15 15:33:34 | Ok to proxy
01-28-15 15:33:34 | Proxy complete
```

when testing this request ``` http://localhost/work/dev/resource-proxy/PHP/proxy.php?http://services.arcgisonline.com/ArcGIS/rest/services?f=json ```  in the log.  Also make sure you have a ```*``` character this in the ```allowedReferers``` section of the config, as this allows for simple testing like  ``` http://localhost/work/dev/resource-proxy/PHP/proxy.php?http://services.arcgisonline.com/ArcGIS/rest/services?f=json ```

**Comment by kodokpintar** [comment]
@jgravois  I tried two things, first I didn't create proxy_log.log file, I got from package (downloaded). Second, I removed the proxy_log.log file from the folder ...\resource-proxy-1.0\PHP\. It didn't create that file when I ran this,

http://localhost/dev/resource-proxy-1.0/PHP/proxy.php?ping. 

It threw me this error:

{"error":{"code":412,"details":["Detected malformed 'logFile' in the configuration file.  Make sure this app has write permissions to log file specified in the configuration file.  The server does not meet one of the preconditions that the requester put on the request."],"message":"Proxy failed due to configuration error."}}


@phpmaps  I meant with there is no issue on permission was I can manually create or delete files on the folder ...\resource-proxy-1.0\PHP\ and also I checked the permission of IIS_IUSRS, it has full control.

FYI,
here is my proxy.config

<?xml version="1.0" encoding="utf-8" ?>
<ProxyConfig allowedReferers="*"
             logFile="proxy_log.log"
             mustMatch="true">
    <serverUrls>
        <serverUrl url="http://services.arcgisonline.com"
                   matchAll="true"/>
    </serverUrls>
</ProxyConfig>



here is my .htaccess

<Files ~ "\.sqlite$">
    Order allow,deny
    Deny from All
</Files>

<Files ~ "\.config$">
    Order allow,deny
    Deny from all
</Files>

<Files ~ "\.log$">
    Order allow,deny
    Deny from all
</Files>

Thank you!

**Comment by hgonzago** [comment]
If you get this error, go into the folder housing the proxy files and create a new empty proxy_log.log file. Make sure that this file and the folder containing all the proxy files (including log) are read/write. To test this, you can run the proxy verification test to make certain.

**Comment by kodokpintar** [comment]
I ran the proxy-verification.php but still It didn't return any result. I can see from my browser - Developer tools that it returned 200 OK.

**Comment by phpmaps** [comment]
@kodokpintar  - when you ran the proxy-verification.php file by opening it in the web browser.. did it display raw code or display a web page with formatted HTML.  If the former that would result in 200 OK but indicate that your server is not provisioned with PHP.  What version of PHP are you running?

**Comment by kodokpintar** [comment]
@phpmaps it didn't display any raw code, it just displayed blank page.
I could run check.php and came up with result, PHP version 5.6.5 and all information, like: configuration, bcmath, calendar, cgi-fcgi, core, ctype, etc

**Comment by phpmaps** [comment]
@kodokpintar - Were you running check.php out of the same web directory as the proxy.php file?  I suggest this create a file called phpinfo.php and include below few lines of code.  Next put the new phpinfo.php file in the same directory as proxy-verification.php and proxy.php files.  Then run phpinfo.php from your web browser - and take a screen grab and share it.

````
<?php
phpinfo();
?>
````

Proxy-verification.php should not result in a blank page, so we need to understand what's messed up.

**Comment by kodokpintar** [comment]
@phpmaps  Please find below the screenshots from phpinfo.php. Thanks!

![image](https://cloud.githubusercontent.com/assets/10455329/5983538/c5bd6432-a893-11e4-9507-9e53fe4c5002.png)
![image](https://cloud.githubusercontent.com/assets/10455329/5983500/8260c24c-a893-11e4-8af1-214eaafe602c.png)
![image](https://cloud.githubusercontent.com/assets/10455329/5983505/8c035bb6-a893-11e4-997c-fd3c6465daa0.png)
![image](https://cloud.githubusercontent.com/assets/10455329/5983509/91b5ee16-a893-11e4-92a4-342375d8a6ea.png)

**Comment by kodokpintar** [comment]
Here is the error when I ran Test Proxy

![image](https://cloud.githubusercontent.com/assets/10455329/5983584/467a99be-a894-11e4-90b9-e4042ba2172f.png)

**Comment by phpmaps** [comment]
I am not seeing PDO_Sqlite and cURL.  Do you have PDO Sqlite and cURL as part of your PHP configuration?  As for this last screen shot showing the error, try setting ````mustMatch```` to ````false```` and make sure you have a ````*```` in the ````allowedReferers```` property and then share your proxy.config file in this thread.

We are getting a little away for the initial issues which was about the missing log file.

The best news, I can offer you right now - is it looks like you server does support PHP.

Maybe something happened causing implementation on IIS to break...  I will try an reproduce on a Windows machine.  @kodokpintar  please do follow up regarding to the PDO Sqlite, cURL and the proxy.config settings when you get time.  It's very strange that the proxy will send you the above message but not display the proxy-verification.php page.

**Comment by kodokpintar** [comment]
@phpmaps please find below two screenshots that showing:
1. pdo_sqlite and curl DLLson PHP extn folder.
![image](https://cloud.githubusercontent.com/assets/10455329/6003120/9ccd19aa-aabe-11e4-8a9e-adf112b7e866.png)

2. proxy.config configuration
![image](https://cloud.githubusercontent.com/assets/10455329/6003139/ba744eba-aabe-11e4-8bc3-d45c1c2fc676.png)


yes, it's strange. I can run php version and other php files but whenever trying to use resource-proxy, it always throw error.
Thank you.

**Comment by phpmaps** [comment]
Thanks that is good feedback.  I went and tested on Windows.  My IIS directory security looked like this in the folder containing the proxy.php file. 

![image](https://cloud.githubusercontent.com/assets/2159502/6011408/422dd868-aaf0-11e4-8759-e12fb5db96ac.png)

Notice IIS_IUSRS has write access.  When it did not have write access - I saw the problem you reported involving the log.  Additionally, I had no problem seeing the proxy-verification.php file.  However, I am seeing an issue on Windows IIS with the "mustMatch" property.  However if your configuration is setup where ````matchAll=false``` everything should just work.   In short with IIS_USRS having write access and ````matchAll=false``` you should 1) see that a log file gets created and 2) be able to proxy this request .

```` http://localhost/dev/proxy-log/PHP/proxy.php?http://services.arcgisonline.com/arcgis/rest/services?f=pjson  ````

Perhaps try downloading the latest resource-proxy.  I tested on today's and today's in new.

**Comment by kodokpintar** [comment]
@phpmaps  THANKS SO MUCH!!! Finally it works!. Two things that I would like to share from my issue:
1. Permission on IIS, somehow it didn't like the permission although I verified IIS_IUSRS had full permission, I made the change on IIS by Adding Web Site (PHP Application) and connect as administrator in app pool.

2. As you suggested to enable cURL and Sqlite by removing semicolons AND I also added path where the DLLs reside. e.g. c:\PHP\ext\php_curl.dll.

Again, Thank you very much for your valuable inputs.
Have a good day!
